src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,401 @@
+    inherit Equinox.Category<'event, 'state, 'context>(resolveInner, empty)
+    new (   context : MessageDbContext, codec : FsCodec.IEventCodec<_, _, 'context>, fold, initial,
+            // Caching can be overkill for EventStore esp considering the degree to which its intrinsic caching is a first class feature
+            // e.g., A key benefit is that reads of streams more than a few pages long get completed in constant time after the initial load
+            [<O; D(null)>]?caching) =
+        let inner = Category<'event, 'state, 'context>(context, codec)
+        let readCacheOption =
+            match caching with
+            | None -> None
+            | Some (CachingStrategy.SlidingWindow (cache, _))
+            | Some (CachingStrategy.FixedTimeSpan (cache, _)) -> Some (cache, null)
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, _, prefix)) -> Some (cache, prefix)
+        let folder = Folder<'event, 'state, 'context>(inner, fold, initial, ?readCache = readCacheOption)
+        let category : ICategory<_, _, 'context> =
+            match caching with
+            | None -> folder :> _
+            | Some (CachingStrategy.SlidingWindow (cache, window)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache null window folder Token.supersedes
+            | Some (CachingStrategy.FixedTimeSpan (cache, period)) ->
+                Caching.applyCacheUpdatesWithFixedTimeSpan cache null period folder Token.supersedes
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, window, prefix)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache prefix window folder Token.supersedes
+        let resolveInner streamIds = struct (category, FsCodec.StreamName.Internal.ofCategoryAndStreamId streamIds, ValueNone)
+        let empty = struct (context.TokenEmpty, initial)
+        MessageDbCategory(resolveInner, empty)
+
+
+type MessageDbConnector(
+    connectionString: string, ?readOnlyConnectionString: string,
+    [<O; D(null)>]?readRetryPolicy, [<O; D(null)>]?writeRetryPolicy) =
+        member _.Connect() =
+            let connectToDb connectionString () =
+                task {
+                    let conn = new NpgsqlConnection(connectionString)
+                    do! conn.OpenAsync()
+                    return conn
+                }
+            let writeConnection = connectToDb connectionString
+            let readConnection =
+                match readOnlyConnectionString with
+                | Some connectionString -> connectToDb connectionString
+                | None -> writeConnection
+            MessageDbClient(writeConnection), MessageDbClient(readConnection)
+
+        member x.Establish() : Task<MessageDbConnection> = task {
+            let write, read = x.Connect()
+            return MessageDbConnection(readConnection=read, writeConnection=write, ?readRetryPolicy=readRetryPolicy, ?writeRetryPolicy=writeRetryPolicy)
```

Comment:
```suggestion
            return MessageDbConnection(readConnection = read, writeConnection = write, ?readRetryPolicy = readRetryPolicy, ?writeRetryPolicy = writeRetryPolicy)
```
Spaces around `=` and `:` in most cases (feel free to update editorconfig if its not in there)